src/Network/Request.php

```diff
@@ -1359,4 +1360,45 @@ public function offsetUnset($name)
     {
         unset($this->params[$name]);
     }
+
+    /**
+     * Returns string representation of the current request in pluginDot syntax
+     *
+     * @params array $options Valid keys are `prefix`, `action`, `pass`, and `query`
+     * @return string Path
+     */
+    public function toPath($options = []) {
+        $prefix = $plugin = $controller = $action = $pass = $query = null;
+        if ($this->plugin) {
+            $plugin = $this->plugin . '.';
+        }
+        if ($prefix = $this->param('prefix')) {
+            $controller = Inflector::camelize($prefix) . '/';
+        }
+        $controller .= $this->param('controller') . 'Controller';
+
+        if (!empty($options['action'])) {
+            $action = '/' . $this->param('action');
+        }
+
+        if (!empty($options['pass']) && $passedArgs = $this->param('pass')) {
+            $pass = '/' . implode('/', $passedArgs);
+        }
+
+        if (!empty($options['query']) && $this->query) {
+            $query = '?' . http_build_query($this->query);
+        }
+
+        return $plugin . $controller . $action . $pass . $query;
```

Comment:
It's not really a URL, instead it's closer to our plugin dot syntax, eg: `$this->belongsTo('Croogo/Users.Roles');`

Now, I wanted to have a 'unique` way to identify a request by a string, so that I can do something like:

```php
Configure::write('EnableTranslations.actions', [
  'Croogo/Nodes.Admin/NodesController/index' => [ ... ],
  'Croogo/Users.Admin/UsersController/index' => [ ... ],
  'Xintesa/Users.Admin/UsersController/index' => [ ... ],
]);
```

When croogo detects a request matches one of the above 'path', it will enable the Translation functionality.

The detection is done by something like:

```php
$shouldEnableTranslation = array_key_exists(
    $this->request->toActionPath(),
    Configure::read('EnableTranslation.actions')
)```